src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Transcoding.cs

```diff
@@ -1,339 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-
-using System.Buffers;
-using System.Diagnostics;
-using System.Runtime.CompilerServices;
-using System.Runtime.InteropServices;
-
-namespace System.Text.Json
-{
-    internal static partial class JsonWriterHelper
-    {
-        // TODO: Replace this with publicly shipping implementation: https://github.com/dotnet/runtime/issues/28204
-        /// <summary>
-        /// Converts a span containing a sequence of UTF-16 bytes into UTF-8 bytes.
-        ///
-        /// This method will consume as many of the input bytes as possible.
-        ///
-        /// On successful exit, the entire input was consumed and encoded successfully. In this case, <paramref name="bytesConsumed"/> will be
-        /// equal to the length of the <paramref name="utf16Source"/> and <paramref name="bytesWritten"/> will equal the total number of bytes written to
-        /// the <paramref name="utf8Destination"/>.
-        /// </summary>
-        /// <param name="utf16Source">A span containing a sequence of UTF-16 bytes.</param>
-        /// <param name="utf8Destination">A span to write the UTF-8 bytes into.</param>
-        /// <param name="bytesConsumed">On exit, contains the number of bytes that were consumed from the <paramref name="utf16Source"/>.</param>
-        /// <param name="bytesWritten">On exit, contains the number of bytes written to <paramref name="utf8Destination"/></param>
-        /// <returns>A <see cref="OperationStatus"/> value representing the state of the conversion.</returns>
-        public static unsafe OperationStatus ToUtf8(ReadOnlySpan<byte> utf16Source, Span<byte> utf8Destination, out int bytesConsumed, out int bytesWritten)
-        {
-            //
-            //
-            // KEEP THIS IMPLEMENTATION IN SYNC WITH https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Text/UTF8Encoding.cs#L845
```

Comment:
I actually wasn't aware we had this method until you brought it up, and I must say I'm glad to see it go. Thanks :-)